libr/core/disasm.c

```diff
@@ -6583,3 +6583,186 @@ R_API int r_core_disasm_pdi(RCore *core, int nb_opcodes, int nb_bytes, int fmt)
 	r_core_seek (core, old_offset, true);
 	return err;
 }
+
+R_API bool r_core_disasm_pde(RCore *core, int nb_opcodes, int mode) {
+	PJ *pj = NULL;
+	if (mode == R_MODE_JSON) {
+		pj = pj_new ();
+		if (!pj) {
+			return false;
+		}
+		pj_a (pj);
+	}
+	if (!core->anal->esil) {
+		r_core_cmd0 (core, "aei");
+		if (!r_config_get_i (core->config, "cfg.debug")) {
+			r_core_cmd0 (core, "aeim");
+		}
+	}
+	const ut64 ocoffset = core->offset;
+	const ut64 ocbsize = core->blocksize;
+	RAnalEsil *esil = core->anal->esil;
+	RReg *reg = core->anal->reg;
+	RList *ocache = core->io->cache;
+	RCache *ocacheb = core->io->buffer;
+	const int ocached = core->io->cached;
+	const bool ocachec = r_config_get_i (core->config, "io.cache");
+	const bool ojmp_lines = r_config_get_i (core->config, "asm.lines");
+	const bool hide_branches = r_config_get_i (core->config, "asm.followcf.hide");
+	if (ocacheb && ocache) {
+		RCache *c = R_NEW0 (RCache);
```

Comment:
what about having  `r_cache_new` and pass the buf+len+base as arguments? that will save some errors, lines and there will be no need to call r_cache_free if the malloc fails or call memcpy at all from the caller